src/implementing_new_feature.md

```diff
@@ -0,0 +1,135 @@
+
+```rust, ignore
+    @rfcbot fcp merge
+```
+
+or
+
+```rust, ignore
+    @rfcbot fcp close
+```
+
+That will cause rfcbot to comment its ticky-box comment.
+
+## The logistics of writing features
+
+There are a few "logistic" hoops you might need to go through in order to implement a feature in a working way.
+The more boring details are listed in the Rust repository's [CONTRIBUTING.md], so I'll just link to it here. You should try to at least skim it fully - it contains a bunch of nice tips.
+
+### Warning Cycles
+
+In some cases, a feature or bugfix might break some existing programs in some edge cases. In that case, you might want to do a crater run to assess the impact and possibly add a warning cycle, following the [rustc bug-fix procedure].
+
+### Stability
+
+We [value the stability of Rust]. Code that works and runs on stable should (mostly) not break. Because of that, we don't want to release a feature to the world with only team consensus and code review - we want to gain real-world experience on using that feature on nightly, and we might want to change the feature based on that experience.
+
+To allow for that, we must make sure users don't accidentally depend on that new feature - otherwise, especially if experimentation takes time or is delayed and the feature takes the trains to stable, it would end up de facto stable and we'll not be able to make changes in it without breaking people's code.
+
+The way we do that is that we make sure all new features are feature gated - they can't be used without a enabling a feature gate (#[feature(foo)]), which can't be done in a stable/beta compiler. See the [stability in code] section for the technical details.
+
+Eventually, after we gain enough experience using the feature, make the necessary changes, and are satisfied, we expose it to the world using the stabilization process described [here]. Until then, the feature is not set in stone: every part of the feature can be changed, or the feature might be completely rewritten or removed. Features are not supposed to gain tenure by being unstable and unchanged for a year.
+
+### Tracking Issues
+
+To keep track of the status of an unstable feature, the experience we get while using it on nightly, and of the concerns that block its stabilization, every feature-gate needs a tracking issue.
+
+General discussions about the feature should be done on the tracking issue.
+
+For features that have an RFC, you should use the RFC's tracking issue for the feature.
+
+For other features, you'll have to make a tracking issue for that feature. The issue title should be "Tracking issue for YOUR FEATURE" and it should have the `B-unstable` & `C-tracking-issue` tags, along with the tag for your subteam (e.g. `T-compiler` if this is a compiler feature).
```

Comment:
Please remove the second half of this sentence (starting from "and it should have...")

Reply:
done